File: spec/java/src/io/kaitai/struct/spec/TestSwitchManualStrElse.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.SwitchManualStrElse;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestSwitchManualStrElse extends CommonSpec {
    @Test
    public void testSwitchManualStrElse() throws Exception {
        SwitchManualStrElse r = SwitchManualStrElse.fromFile(SRC_DIR + "switch_opcodes2.bin");

        assertEquals(r.opcodes().size(), 4);

        assertEquals(r.opcodes().get(0).code(), "S");
        assertEquals(((SwitchManualStrElse.Opcode.Strval) r.opcodes().get(0).body()).value(), "foo");

        assertEquals(r.opcodes().get(1).code(), "X");
        assertEquals(((SwitchManualStrElse.Opcode.Noneval) r.opcodes().get(1).body()).filler(), 0x42);

        assertEquals(r.opcodes().get(2).code(), "Y");
        assertEquals(((SwitchManualStrElse.Opcode.Noneval) r.opcodes().get(2).body()).filler(), 0xcafe);

        assertEquals(r.opcodes().get(3).code(), "I");
        assertEquals(((SwitchManualStrElse.Opcode.Intval) r.opcodes().get(3).body()).value(), 7);
    }
}

File: spec/javascript/test_switch_manual_str_else.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('SwitchManualStrElse', 'src/switch_opcodes2.bin', function(r) {
  assert.equal(r.opcodes.length, 4);

  assert.equal(r.opcodes[0].code, 'S');
  assert.equal(r.opcodes[0].body.value, 'foo');

  assert.equal(r.opcodes[1].code, 'X');
  assert.equal(r.opcodes[1].body.filler, 0x42);

  assert.equal(r.opcodes[2].code, 'Y');
  assert.equal(r.opcodes[2].body.filler, 0xcafe);

  assert.equal(r.opcodes[3].code, 'I');
  assert.equal(r.opcodes[3].body.value, 7);
});
